python/pyarrow/_csv.pyx

```diff
@@ -296,10 +297,31 @@ cdef class ParseOptions:
         out.options = options
         return out
 
-    def __reduce__(self):
-        return ParseOptions, (self.delimiter, self.quote_char,
-                              self.double_quote, self.escape_char,
-                              self.newlines_in_values, self.ignore_empty_lines)
+    def __getstate__(self):
+        return (self.delimiter, self.quote_char, self.double_quote,
+                self.escape_char, self.newlines_in_values,
+                self.ignore_empty_lines)
+
+    def __setstate__(self, state):
+        (self.delimiter, self.quote_char, self.double_quote,
+         self.escape_char, self.newlines_in_values,
+         self.ignore_empty_lines) = state
+
+
+cdef class _ISO8601:
+    """
+    A special object indicating ISO-8601 parsing.
```

Comment:
Alternative could also be to use a string for this? (so you could do like `["ISO-8601", "%Y/%m/%d"]`) 
I understand that it is cleaner to use a separate object since "ISO-8601" is strictly speaking not a format string, but it would that users need to use yet another object (no strong objection to the current approach, though, to be clear).

Reply:
It's easy to misspell "ISO-8601" (for example forget the hyphen, or use lowercase) and then get an entirely different behaviour :-) That's why I went for the dedicated object (if you misspell it, you get an exception).

Reply:
Good ux feature, I'll take note.